client/scripts/views/components/new_thread_form.ts

```diff
@@ -257,40 +361,56 @@ export const NewThreadForm: m.Component<{
               intent: 'primary',
               label: 'Create link',
               name: 'submit',
-              onclick: (e) => {
-                if (!vnode.state.error.url && !detectURL(vnode.state.form.url)) {
-                  vnode.state.error.url = 'Must provide a valid URL.';
-                }
-                if (!Object.values(vnode.state.error).length) {
-                  newLink(vnode.state.form, vnode.state.quillEditorState, author);
-                }
-                if (vnode.attrs.isModal && !vnode.state.error) {
-                  $(e.target).trigger('modalcomplete');
-                  setTimeout(() => {
-                    $(e.target).trigger('modalexit');
-                  }, 0);
+              onclick: async (e) => {
+                vnode.state.saving = true;
+                if (!detectURL(vnode.state.form.url)) {
+                  notifyError('Must provide a valid URL.');
+                } else {
+                  if (!vnode.state.form.linkTitle) {
+                    vnode.state.form.linkTitle = ($(vnode.dom).find('input[name=\'new-link-title\'').val() as string);
+                  }
+                  try {
+                    await newLink(vnode.state.form, vnode.state.quillEditorState, author);
+                    vnode.state.saving = false;
+                    if (isModal) {
+                      $(e.target).trigger('modalcomplete');
+                      setTimeout(() => {
+                        $(e.target).trigger('modalexit');
+                        clearLocalStorage(PostType.Link);
+                      }, 0);
+                    } else {
+                      clearLocalStorage(PostType.Discussion);
+                    }
+                  } catch (err) {
+                    vnode.state.saving = false;
+                    notifyError(err.message);
+                  }
                 }
               },
             }),
           ]),
-          error
-            && (typeof error === 'string' || Object.keys(error).length)
-            ? m('.error-message', [
-              (typeof error === 'string')
-                ? m('span', error)
-                : Object.values(error).map((val) => m('span', `${val} `)),
-            ])
-            : m('.error-placeholder'),
+          // error
+          //   && (typeof error === 'string' || Object.keys(error).length)
+          //   ? m('.error-message', [
+          //     (typeof error === 'string')
+          //       ? m('span', error)
+          //       : Object.values(error).map((val) => m('span', `${val} `)),
+          //   ])
+          //   : m('.error-placeholder'),
```

Comment:
why commented out?

did we lose the ability to show errors?

Reply:
The form was having problems rendering the errors at the bottom of the modal. Oftentimes those errors would be invisible to the user. In addition, there was a lot of very messy error creation/deletion logic that was overgrown. Replacing with notifyError fixed both those problems in (my) original error-handling design.

Reply:
Okay, we can use notifyError for the form...we should remove this though.